src/extension/background.js

```diff
@@ -576,6 +578,23 @@ const internalActions = {
     chrome.contextMenus.onClicked.addListener(async ({ menuItemId }, tab) => {
       if (!tab.url) return;
       internalActions[menuItemId](tab);
+      await chrome.scripting.executeScript({
+        target: { tabId: tab.id },
+        func: async () => {
+          try {
+            const sampleRUM = await import(chrome.runtime.getURL('rum.js'));
+
+            // Ensure window.hlx and window.hlx.sidekick exists
+            window.hlx = window.hlx || {};
+            window.hlx.sidekick = window.hlx.sidekick || { location: window.location };
+
+            const checkpoint = `sidekick:context-menu:${menuItemId}`;
+            sampleRUM(checkpoint, { source: window.location.href });
+          } catch (e) {
+            log.error('Error injecting rum', e);
```

Comment:
Since this code will be executed in the context of the page, you can't use variables (like `log`) from the context of the service worker here. use `console.log` to inspect any errors.